acceptance/detector_test.go

```diff
@@ -0,0 +1,197 @@
+		})
+	})
+
+	when("no buildpack group passed detection", func() {
+		it("errors", func() {
+			cmd := exec.Command("docker", "run", "--rm", "--user", userID, "--env", "CNB_ORDER_PATH=/cnb/orders/empty_order.toml", detectImage)
+			output, err := cmd.CombinedOutput()
+			h.AssertNotNil(t, err)
+			expected := "No buildpack groups passed detection."
+			h.AssertStringContains(t, string(output), expected)
+		})
+	})
+
+	when("there is a buildpack group that pass detection", func() {
+		var copyDir, containerName string
+
+		it.Before(func() {
+			containerName = "test-container-" + h.RandString(10)
+			var err error
+			copyDir, err = ioutil.TempDir("", "test-docker-copy-")
+			h.AssertNil(t, err)
+		})
+
+		it.After(func() {
+			if h.DockerContainerExists(t, containerName) {
+				h.Run(t, exec.Command("docker", "rm", containerName))
+			}
+			os.RemoveAll(copyDir)
+		})
+
+		it("writes group.toml and plan.toml", func() {
+			h.DockerRunAndCopy(t,
+				containerName,
+				copyDir,
+				detectImage,
+				"/layers",
+				h.WithFlags("--user", userID,
+					"--env", "CNB_ORDER_PATH=/cnb/orders/simple_order.toml",
+				),
+				h.WithArgs(),
+			)
+
+			// check group.toml
+			tempGroupToml := filepath.Join(copyDir, "layers", "group.toml")
+			groupContents, err := ioutil.ReadFile(tempGroupToml)
+			h.AssertNil(t, err)
+			h.AssertEq(t, len(groupContents) > 0, true)
+			var buildpackGroup lifecycle.BuildpackGroup
+			_, err = toml.Decode(string(groupContents), &buildpackGroup)
+			h.AssertNil(t, err)
+			h.AssertEq(t, buildpackGroup.Group[0].ID, "simple_buildpack")
+			h.AssertEq(t, buildpackGroup.Group[0].Version, "simple_buildpack_version")
+
+			// check plan.toml - should be empty
+			tempPlanToml := filepath.Join(copyDir, "layers", "plan.toml")
+			planContents, err := ioutil.ReadFile(tempPlanToml)
+			h.AssertNil(t, err)
+			h.AssertEq(t, len(planContents) > 0, true)
+			var buildPlan lifecycle.BuildPlan
+			_, err = toml.Decode(string(planContents), &buildPlan)
+			h.AssertNil(t, err)
+			h.AssertEq(t, buildPlan.Entries[0].Providers[0].ID, "simple_buildpack")
+			h.AssertEq(t, buildPlan.Entries[0].Providers[0].Version, "simple_buildpack_version")
+			h.AssertEq(t, buildPlan.Entries[0].Requires[0].Name, "some_requirement")
+			h.AssertEq(t, buildPlan.Entries[0].Requires[0].Metadata["some_metadata_key"], "some_metadata_val")
+			h.AssertEq(t, buildPlan.Entries[0].Requires[0].Metadata["version"], "0.1")
+		})
+	})
+
+	when("environment variables are provided for buildpack and app directories and for the output files", func() {
+		var copyDir, containerName string
+
+		it.Before(func() {
+			containerName = "test-container-" + h.RandString(10)
+			var err error
+			copyDir, err = ioutil.TempDir("", "test-docker-copy-")
+			h.AssertNil(t, err)
+		})
+
+		it.After(func() {
+			if h.DockerContainerExists(t, containerName) {
+				h.Run(t, exec.Command("docker", "rm", containerName))
+			}
+			os.RemoveAll(copyDir)
+		})
```

Comment:
Similar to above comment re: analyzer acceptance, it would be great not to have to repeat this setup for every test.

Reply:
Please review my last commit.